modules/azure/src/main/scala/com.snowplowanalytics.snowplow.lakes/AzureApp.scala

```diff
@@ -35,5 +39,23 @@ object AzureApp extends LoaderApp[KafkaSourceConfig, KafkaSinkConfig](BuildInfo)
 
   override def badSink: SinkProvider = KafkaSink.resource(_, classTag[SinkAuthHandler])
 
-  override def isDestinationSetupError: DestinationSetupErrorCheck = TableFormatSetupError.check
+  override def isDestinationSetupError: DestinationSetupErrorCheck = {
+    // Authentication issue (wrong OAuth endpoint, wrong client id, wrong secret..)
+    case e: AbfsRestOperationException if e.getStatusCode == -1 =>
+      e.getErrorMessage()
```

Comment:
Example with wrong endpoint:
```
Failed to create events table: Auth failure: HTTP Error 400; url='https://login.microsoftonline.com/0cf615b7-3054-4920-9991-58e992424671/oauth2/token' AADToken: HTTP connection to https://login.microsoftonline.com/0cf615b7-3054-4920-9991-58e992424671/oauth2/token failed for getting token from AzureAD.; requestId='cb17fcea-c326-4716-9c1f-f92c87765200'; contentType='application/json; charset=utf-8'; response '{"error":"invalid_request","error_description":"AADSTS90002: Tenant '0cf615b7-3054-4920-9991-58e992424671' not found. Check to make sure you have the correct tenant ID and are signing into the correct cloud. Check with your subscription administrator, this may happen if there are no active subscriptions for the tenant. Trace ID: cb17fcea-c326-4716-9c1f-f92c87765200 Correlation ID: 1e7caf23-be53-4aa8-991b-a458fd187357 Timestamp: 2024-10-18 13:52:56Z","error_codes":[90002],"timestamp":"2024-10-18 13:52:56Z","trace_id":"cb17fcea-c326-4716-9c1f-f92c87765200","correlation_id":"1e7caf23-be53-4aa8-991b-a458fd187357","error_uri":"https://login.microsoftonline.com/error?code=90002"}'
```
Wrong client id:
```
Failed to create events table: Auth failure: HTTP Error 400; url='https://login.microsoftonline.com/0cf615b7-3054-4920-9991-58e992424670/oauth2/token' AADToken: HTTP connection to https://login.microsoftonline.com/0cf615b7-3054-4920-9991-58e992424670/oauth2/token failed for getting token from AzureAD.; requestId='d594c7e6-60cb-43ab-b89b-687a87c31600'; contentType='application/json; charset=utf-8'; response '{"error":"unauthorized_client","error_description":"AADSTS700016: Application with identifier 'b1874e74-163f-4ca3-9551-16aa5c6460eb' was not found in the directory 'Snowplow'. This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You may have sent your authentication request to the wrong tenant. Trace ID: d594c7e6-60cb-43ab-b89b-687a87c31600 Correlation ID: 899535f2-64a6-49ef-a256-0f3a75bf81a9 Timestamp: 2024-10-18 13:54:14Z","error_codes":[700016],"timestamp":"2024-10-18 13:54:14Z","trace_id":"d594c7e6-60cb-43ab-b89b-687a87c31600","correlation_id":"899535f2-64a6-49ef-a256-0f3a75bf81a9","error_uri":"https://login.microsoftonline.com/error?code=700016"}'
```
Wrong secret:
```
Failed to create events table: Auth failure: HTTP Error 401; url='https://login.microsoftonline.com/0cf615b7-3054-4920-9991-58e992424670/oauth2/token' AADToken: HTTP connection to https://login.microsoftonline.com/0cf615b7-3054-4920-9991-58e992424670/oauth2/token failed for getting token from AzureAD.; requestId='7baa91ac-4829-4d01-ad6a-bc5bc55b2800'; contentType='application/json; charset=utf-8'; response '{"error":"invalid_client","error_description":"AADSTS7000215: Invalid client secret provided. Ensure the secret being sent in the request is the client secret value, not the client secret ID, for a secret added to app 'b1874e74-163f-4ca3-9551-16aa5c6460ea'. Trace ID: 7baa91ac-4829-4d01-ad6a-bc5bc55b2800 Correlation ID: 05cc7746-12cc-4d48-9495-203803405b89 Timestamp: 2024-10-18 13:55:37Z","error_codes":[7000215],"timestamp":"2024-10-18 13:55:37Z","trace_id":"7baa91ac-4829-4d01-ad6a-bc5bc55b2800","correlation_id":"05cc7746-12cc-4d48-9495-203803405b89","error_uri":"https://login.microsoftonline.com/error?code=7000215"}'
```